File: bindings/libotp-php/src/PHPOTP/Authenticator/Authenticator.php
<?php

namespace PHPOTP\Authenticator;


class Authenticator{


	/**
	* @var mixed
	*/
	protected $secret;

    /**
    * @var string
    */
	protected $algo;

	public function __construct($secret,$hash_algo = "sha1"){
		
		$this->setSecret($secret);

		$this->setAlgo($hash_algo);

	}


	/**
	*
	* @param mixed $data
	* @return mixed binary data
	*/
	public function hash($data){
		return hash_hmac($this->getAlgo(), $data , $this->getSecret(),true);
	}




    /**
     * Gets the value of secret.
     *
     * @return mixed
     */
    public function getSecret()
    {
        return $this->secret;
    }
    
    /**
     * Sets the value of secret.
     *
     * @param mixed $secret the secret
     *     
     */
    public function setSecret($secret)
    {
        $this->secret = $secret;
    }





    /**
     * Gets the value of algo.
     *
     * @return string
     */
    public function getAlgo()
    {
        return $this->algo;
    }
    
    /**
     * Sets the value of algo.
     *
     * @param string $algo the algo
     *
     */
    public function setAlgo($algo)
    {
        $this->algo = $algo;
    }
}

File: bindings/libotp-java/src/main/java/com/libotp/Authenticator/Authenticator.java
package com.libotp.Authenticator;

import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;
import java.nio.ByteBuffer;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;

/**
 * Created by luissantos on 2/4/14.
 */
public class Authenticator {

    public static final String HMAC_SHA1_ALGORITHM = "HmacSHA1";

    protected byte[] secret;

    protected String algo;


    public Authenticator(byte[] secret) {
        this(secret,HMAC_SHA1_ALGORITHM);
    }

    public Authenticator(byte[] secret,String algo) {
        this.setSecret(secret);
        this.setAlgo(algo);
    }

    public byte[] getSecret() {
        return secret;
    }

    public void setSecret(byte[] secret) {
        this.secret = secret;
    }

    public String getAlgo() {
        return algo;
    }

    public void setAlgo(String algo) {
        this.algo = algo;
    }

    /**
     *
     * @param data
     * @return
     * @throws NoSuchAlgorithmException
     * @throws InvalidKeyException
     */
    public byte[] hash(byte[] data) throws NoSuchAlgorithmException, InvalidKeyException {

        SecretKeySpec signingKey = new SecretKeySpec(getSecret(), HMAC_SHA1_ALGORITHM );

        Mac mac = Mac.getInstance(HMAC_SHA1_ALGORITHM);
        mac.init(signingKey);


        return mac.doFinal(data);
    }

    /**
     *
     * @param data
     * @return
     * @throws NoSuchAlgorithmException
     * @throws InvalidKeyException
     */
    public byte[] hash(String data) throws NoSuchAlgorithmException, InvalidKeyException {
       return this.hash(data.getBytes());
    }

    /**
     *
     * @param data
     * @return
     * @throws NoSuchAlgorithmException
     * @throws InvalidKeyException
     */
    public byte[] hash(Integer data) throws NoSuchAlgorithmException, InvalidKeyException {

        byte[] integer_data =  ByteBuffer.allocate(4).putInt(data).array();

               
        return this.hash(integer_data);
    }


}
